### ISSUE TITLE
Handle SIGHUP

### ISSUE THREAD

**Issue by Ppjet6** [restic/rest-server#80]
This is related to https://forum.restic.net/t/rest-server-crashes-or-exits-unexpectedly/1073/4

version: 0.9.7 or dfe9755ed093bbdb9085b62218aeae4f33c939a0 (master)

## Steps to reproduce

- Start rest-server
- `kill -HUP <pid>`

## What happens

rest-server exits.

## What I expect

At least not to die, (ignoring the signal?). Reloading log files would be nice, and that seems to be the usual behaviour for this signal.

Some background, I discovered this because rest-server was unexpectedly dying, and logfiles would be empty. It appears that synology, the box we're running it on, generated a logrotate configuration for it and was killing it every so often. That wasn't fun.

I deactivated it for now, but handling SIGHUP would be nice. Rotating log files can be useful :)

**Comment by mholt** [comment]
Have you tried with `nohup`?

**Comment by Ppjet6** [comment]
Right. I have another workaround for the moment.

Can we repurpose this issue then to have `SIGHUP` handled entirely, (as in reopening fds etc.)

**Comment by mholt** [comment]
Sure, I think we can do that. So what behavior do you need for the log rotation to work? Just ignore HUP?

**Comment by Ppjet6** [comment]
No, for log rotation to work, we'd need to have restic reopen every fds it has open, so that logrotate can move/archive them.
